src/main/java/tech/blacksource/blacknectar/service/algorithms/JaroWinklerDistance.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2017 BlackSource, LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ 
+package tech.blacksource.blacknectar.service.algorithms;
+
+
+import com.google.common.base.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static tech.sirwellington.alchemy.arguments.Arguments.checkThat;
+import static tech.sirwellington.alchemy.arguments.assertions.StringAssertions.nonEmptyString;
+
+/**
+ *
+ * @author SirWellington
+ */
+final class JaroWinklerDistance implements StringDistanceAlgorithm
```

Comment:
Somehow this code is already in develop, Will just leave it here for now, but watch it next time.